platform/linux-generic/arch/aarch64/odp_random.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_AARCH64_RANDOM_H_
+#define ODP_AARCH64_RANDOM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/spec/random.h>
+#include <odp/autoheader_internal.h>
+
+#include <stdint.h>
+
+#define RNDR_OUTPUT_LEN      8
+
+odp_random_kind_t _odp_random_max_kind_generic(void);
+int32_t _odp_random_true_data_generic(uint8_t *buf, uint32_t len);
+int32_t _odp_random_crypto_data_generic(uint8_t *buf, uint32_t len);
+
+static inline int _odp_random_max_kind(void)
+{
+	if (_ODP_OPENSSL_RAND) {
+#ifdef __ARM_FEATURE_RNG
+		return ODP_RANDOM_TRUE;
+#else
+		return ODP_RANDOM_CRYPTO;
+#endif
+	} else {
+		return ODP_RANDOM_BASIC;
+	}
+}
+
+static inline int _odp_rndr(unsigned long *v)
+{
+	int pass;
+
+	__asm__ volatile("mrs   %0, s3_3_c2_c4_0\n\t"  /* RNDR */
+			 "cset  %w[pass], ne"
+			 : "=r" (*v), [pass] "=r" (pass)
+			 :
+			 : "cc");
+
+	return pass;
+}
+
+static inline int32_t _odp_random_true_data(uint8_t *buf, uint32_t len)
+{
+#ifdef __ARM_FEATURE_RNG
+	if (_ODP_OPENSSL_RAND) {
+		while (!_odp_rndr((unsigned long *)buf))
+			;
+
+		return RNDR_OUTPUT_LEN;
```

Comment:
The function should at least try to generate `len` bytes of random data.

Reply:
RNDR registers gives 64-bit random data so we had fixed the return length to be the same. I am not exactly clear on what change should I do here. Should I add a check for len and RNDR_OUTPUT_LEN or should the change be something else?

Reply:
An application may request for example only 1 byte of random data, so you cannot write directly to the output `buf`. To make application usage easier you could add a loop here and write 64 bits at a time until `len` bytes have been written.

Reply:
Okay, I'll make this change in v2. Thanks!